File: playService/assets/src/common/modal.js
var h = require('snabbdom/h').default

function modal(content, onClose) {
        return h('div.modal.is-active', [
            h('div.modal-background', onClose ? {on: {click: onClose}}: null),
            h('div.modal-content', [
                h('div.box.content', content)
            ]),
             onClose ? h('button.modal-close.is-large', {on: {click: onClose}}) : null
        ])
}

module.exports = modal

File: frontend/src/main/scala/roborace/frontend/pages/components/Modal.scala
package roborace.frontend.pages.components

import snabbdom.{Eventlistener, Node}

// todo move background to components and use that one
object Modal {
  def apply(closeAction: Eventlistener, background: Option[Node] = None)(content: Node*): Node =
    Node("div.modal.is-active")
      .child(
        Node("div.modal-background")
          .event("click", closeAction)
          .style("display", "flex")
          .childOptional(background)
      )
      .child(Node("div.modal-content").child(Node("div.box").child(content)))
}
